cmd/registry/main.go

```diff
@@ -32,10 +33,21 @@ func main() {
 	})
 
 	cmd := cmd.Command()
+
+	// Customize UsageFunc() so that SilenceUsage will hide our extra help info.
+	uf := cmd.UsageFunc()
+	cmd.SetUsageFunc(func(c *cobra.Command) error {
+		if err := uf(c); err != nil {
+			return err
+		}
+		_, err := fmt.Fprint(c.OutOrStderr(), `----
+Need more help?
+https://github.com/apigee/registry/wiki
+`)
+		return err
+	})
+
 	if err := cmd.ExecuteContext(ctx); err != nil {
-		fmt.Println("----")
-		fmt.Println("Need more help?")
-		fmt.Println("https://github.com/apigee/registry/wiki")
```

Comment:
Unfortunately `cmd.SilenceUsage` isn't helpful here. `cmd` is the top-level command, and in our implementation, we set it on the get command. It seemed that the best way to silence this was to add it to the `UsageString()` with the code above.

Reply:
Hmm. I'll have to think about this more, but first blush is that maybe we're not properly handling what should be returned as an error if this is a problem here.

Reply:
"Should all errors print usage?" seems to be the key question for us to answer.

Reply:
The reason for adding the "Need more help" was to provide help to users when they encounter errors, so I would argue that it is good to do so.  For anything that is potentially expected and not an *error*, we should probably handle within the command. In this case, we've decided that NotFound is not actually an error, just a valid result.

Reply:
This is messy. I'm not saying that I want to match this, but the generated `rpc list-` commands return an error when no items match a list. 
```
registry rpc list-apis --parent projects/test/locations/global --filter "api_id == '12'"

Error: no more items in iterator
Usage:
  registry rpc list-apis [flags]
```

Similarly, getting a non-existing resource returns an error in the `rpc` commands.
```
$ registry rpc get-api --name projects/test/locations/global/apis/missing
Error: rpc error: code = NotFound desc = "projects/test/locations/global/apis/missing" not found in database
```

So while I'm not sure that we want to match this, I think not matching it should be for a good reason. Turning off the distracting and irrelevant usage information is one reason (if you ask for a resource that doesn't exist, I don't think you need to be reminded of the usage information, and the usage info gets in the way of seeing what actually happened). I think returning an error would also affect the return value of the `registry` tool, which could be significant in scripts. Do we want to return OK when the `registry` tool is run to `get` something that doesn't exist? I think that might be a bad idea.

Reply:
Rambling on about this, I think we could narrow this to just being about `registry get` (which it is) and we could say that `registry get` is a command that returns something if it exists and nothing if nothing matches (with no error) and that it only errors on bad inputs like invalid filters or ...invalid patterns? If I forget to include "/locations/global" in a full path or leave out the collection identifiers ("apis", "versions", etc), then the underlying Visitor code should detect this and report an error (as it should, I think).

Reply:
My argument was just that the root of problem is that each command may have different interpretations of what an actual "error" should be... and if an error isn't handled and bubbles up to the top level, it should probably go ahead and print usage information for that command to be of help the user. (Which is the default for cobra.) NotFound seems to be in a No Man's Land of "is it an user error or not" and is pretty common... so I'm good with just checking for that and moving on.

Reply:
OK, I agree that we'll treat NotFound as not-an-error in `registry get`, so we can leave everything else as is.